src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1528,1894 @@ interface MPSMatrixMultiplication {
+
+		[Export ("initWithDevice:kernelWidth:kernelHeight:dilationRateX:dilationRateY:strideInPixelsX:strideInPixelsY:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint kernelWidth, nuint kernelHeight, nuint dilationRateX, nuint dilationRateY, nuint strideInPixelsX, nuint strideInPixelsY);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnKernel), Name = "MPSCNNUpsampling")]
+	interface MPSCnnUpsampling
+	{
+		[Export ("scaleFactorX")]
+		double ScaleFactorX { get; }
+
+		[Export ("scaleFactorY")]
+		double ScaleFactorY { get; }
+
+		// inlining ctor from base class
+		[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnUpsampling), Name = "MPSCNNUpsamplingNearest")]
+	interface MPSCnnUpsamplingNearest
+	{
+		[Export ("initWithDevice:integerScaleFactorX:integerScaleFactorY:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint integerScaleFactorX, nuint integerScaleFactorY);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnUpsampling), Name = "MPSCNNUpsamplingBilinear")]
+	interface MPSCnnUpsamplingBilinear
+	{
+		[Export ("initWithDevice:integerScaleFactorX:integerScaleFactorY:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, nuint integerScaleFactorX, nuint integerScaleFactorY);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(NSObject), Name = "MPSRNNDescriptor")]
+	interface MPSRnnDescriptor
+	{
+		[Export ("inputFeatureChannels")]
+		nuint InputFeatureChannels { get; set; }
+
+		[Export ("outputFeatureChannels")]
+		nuint OutputFeatureChannels { get; set; }
+
+		[Export ("useLayerInputUnitTransformMode")]
+		bool UseLayerInputUnitTransformMode { get; set; }
+
+		[Export ("useFloat32Weights")]
+		bool UseFloat32Weights { get; set; }
+
+		[Export ("layerSequenceDirection", ArgumentSemantic.Assign)]
+		MPSRnnSequenceDirection LayerSequenceDirection { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSRnnDescriptor), Name = "MPSRNNSingleGateDescriptor")]
+	interface MPSRnnSingleGateDescriptor
+	{
+		[NullAllowed, Export ("inputWeights", ArgumentSemantic.Retain)]
+		IMPSCnnConvolutionDataSource InputWeights { get; set; }
+
+		[NullAllowed, Export ("recurrentWeights", ArgumentSemantic.Retain)]
+		IMPSCnnConvolutionDataSource RecurrentWeights { get; set; }
+
+		[Static]
+		[Export ("createRNNSingleGateDescriptorWithInputFeatureChannels:outputFeatureChannels:")]
+		MPSRnnSingleGateDescriptor CreateRNNSingleGateDescriptor (nuint inputFeatureChannels, nuint outputFeatureChannels);
```

Comment:
`CreateRnnSingleGateDescriptor` (different case for `Rnn`) to match the class name.